File: mbw/src/main/java/com/mycelium/bequant/kyc/verticalStepper/VerticalStepperItemCircleView.java
package com.mycelium.bequant.kyc.verticalStepper;

import android.annotation.TargetApi;
import android.content.Context;
import android.graphics.drawable.GradientDrawable;
import android.os.Build;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import com.mycelium.wallet.R;

public class VerticalStepperItemCircleView extends FrameLayout {
    private TextView number;

    private ImageView icon;

    public VerticalStepperItemCircleView(Context context) {
        super(context);
        initialize(context);
    }

    public VerticalStepperItemCircleView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initialize(context);
    }

    public VerticalStepperItemCircleView(
            Context context,
            AttributeSet attrs,
            int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initialize(context);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public VerticalStepperItemCircleView(
            Context context,
            AttributeSet attrs,
            int defStyleAttr,
            int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        initialize(context);
    }

    private void initialize(Context context) {
        LayoutInflater.from(context).inflate(
                R.layout.vertical_stepper_view_item_circle,
                this,
                true);

        number = (TextView) findViewById(R.id.vertical_stepper_view_item_circle_number);
        icon = (ImageView) findViewById(R.id.vertical_stepper_view_item_circle_icon);
    }

    public void setBackgroundActive() {
        GradientDrawable drawable = (GradientDrawable) ContextCompat
                .getDrawable(
                        getContext(),
                        R.drawable.vertical_stepper_view_item_circle_active);
        drawable.setColor(Util
                .getThemeColor(getContext(), R.attr.colorAccent));
        setBackgroundResource(R.drawable.vertical_stepper_view_item_circle_active);
    }

    public void setBackgroundInactive() {
        setBackgroundResource(R.drawable.vertical_stepper_view_item_circle_inactive);
    }

    public void setNumber(int value) {
        icon.setVisibility(View.GONE);
        number.setVisibility(View.VISIBLE);
        number.setText(String.valueOf(value));
    }

    public void setIconCheck() {
        setIconResource(R.drawable.ic_vertical_step);
    }

    public void setIconEdit() {
        setIconResource(R.drawable.ic_vertical_step);
    }

    public void setIconResource(int id) {
        number.setVisibility(View.GONE);
        icon.setVisibility(View.VISIBLE);
        icon.setImageResource(id);
    }
}

File: mbw/src/main/java/com/mycelium/bequant/kyc/verticalStepper/VerticalStepperItemCircleView.kt
package com.mycelium.bequant.kyc.verticalStepper

import android.annotation.TargetApi
import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.DrawableRes
import androidx.core.content.ContextCompat
import androidx.core.content.res.ResourcesCompat
import com.mycelium.wallet.R

class VerticalStepperItemCircleView : FrameLayout {
    private var number: TextView? = null
    private var icon: ImageView? = null

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    init {
        LayoutInflater.from(context).inflate(R.layout.vertical_stepper_view_item_circle, this, true)

        number = findViewById<View>(R.id.vertical_stepper_view_item_circle_number) as TextView
        icon = findViewById<View>(R.id.vertical_stepper_view_item_circle_icon) as ImageView
    }

    fun setActive() {
        background = ContextCompat.getDrawable(context, R.drawable.vertical_stepper_view_item_circle_active)
        number?.setTextColor(ResourcesCompat.getColor(resources, R.color.bequant_gray_3, null))
    }

    fun setComplete() {
        background = ContextCompat.getDrawable(context, R.drawable.vertical_stepper_view_item_circle_completed)
    }

    fun setInactive() {
        background = ContextCompat.getDrawable(context, R.drawable.vertical_stepper_view_item_circle_inactive)
        number?.setTextColor(ResourcesCompat.getColor(resources, R.color.white, null))
    }

    fun setError() {
        background = ContextCompat.getDrawable(context, R.drawable.vertical_stepper_view_item_circle_error)
    }
    
    fun setNumber(value: Int) {
        icon!!.visibility = View.GONE
        number!!.visibility = View.VISIBLE
        number!!.text = value.toString()
    }

    fun setIconCheck() {
        setIconResource(R.drawable.ic_vertical_stepper_done)
    }

    fun setIconEdit() {
        setIconResource(R.drawable.ic_vertical_stepper_edit)
    }

    fun setIconResource(id: Int) {
        number!!.visibility = View.GONE
        icon!!.visibility = View.VISIBLE
        icon!!.setImageResource(id)
    }
}